script.js

```diff
@@ -1,4 +1,192 @@
+    //checking if deciding order input is valid
+    var isDiceOrderInputValid = checkIfOrderInputIsValid(input);
+    if (!isDiceOrderInputValid) {
+      return `Invalid input - please enter either 'left' or 'right!`;
+    } else {
+      var playerOneChoice = input;
+    }
+    gameMode = `player two`;
+    if (playerOneChoice == "left") {
+      playerOneChosenOrder = `${playerOneDiceRoll1}${playerOneDiceRoll2}`;
+    }
+    if (playerOneChoice == "right") {
+      playerOneChosenOrder = `${playerOneDiceRoll2}${playerOneDiceRoll1}`;
+    }
+    myOutputValue = `${userNamePlayerOne}, you have chosen **. <br> (shhh🤫...keep it a secret! - Player 2 will now play the game!)`;
+    return myOutputValue;
+  }
+
+  //player two enters name and rolls the dice
+  if (gameMode == `player two`) {
+    numberOfGamesPlayed = numberOfGamesPlayed + 1;
+
+    //input validation for player two username
+    var isUserNameInputValid = checkIfInputIsValid(input);
+    if (!isUserNameInputValid) {
+      return `Invalid input - please enter your name to play the game!`;
+    }
+    userNamePlayerTwo = input;
+    playerTwoDiceRoll1 = diceRoll();
+    playerTwoDiceRoll2 = diceRoll();
+    playerTwoDiceRolls.push(playerTwoDiceRoll1, playerTwoDiceRoll2);
+    myOutputValue = `🎲Hi ${userNamePlayerTwo}, You have rolled dice ${playerTwoDiceRolls}🎲<br> Please indicate the order of dice by choosing "left" or "right" - the biggest number will win!`;
+    gameMode = `player two deciding order`;
+    return myOutputValue;
+  }
+
+  //player two deciding order of dice
+  if (gameMode == `player two deciding order`) {
+    var isDiceOrderInputValid = checkIfOrderInputIsValid(input);
+    if (!isDiceOrderInputValid) {
+      return `Invalid input - please enter either 'left' or 'right!`;
+    } else {
+      var playerTwoChoice = input;
+    }
+    gameMode = `deciding winner`;
+    if (playerTwoChoice == "left") {
+      playerTwoChosenOrder = `${playerTwoDiceRoll1}${playerTwoDiceRoll2}`;
+    }
+    if (playerTwoChoice == "right") {
+      playerTwoChosenOrder = `${playerTwoDiceRoll2}${playerTwoDiceRoll1}`;
+    }
+    myOutputValue = `${userNamePlayerTwo}, you have chosen ${playerTwoChosenOrder} - and the winner is...*drumrolls* 🙌 <br>Click the 'Submit' button to find out!`;
+    return myOutputValue;
+  }
+
+  //showing who is the winner
+  if (gameMode == `deciding winner`) {
+    playerOneRunningScore =
+      playerOneRunningScore + Number(playerOneChosenOrder);
+    playerTwoRunningScore =
+      playerTwoRunningScore + Number(playerTwoChosenOrder);
+
+    gameMode = "player one";
+
+    //general scoreboard statement
+    var scoreboardStatement = `<br><u>Current Scoreboard</u><br>Total Number of Games Played: ${numberOfGamesPlayed}<br>${userNamePlayerOne}'s Dice Roll Score: ${playerOneRunningScore} <br> ${userNamePlayerTwo}'s Dice Roll Score: ${playerTwoRunningScore}`;
+
+    //generating leaderboard order
+    var orderOfLeaderboard = function (
```

Comment:
very good use of helper function to make your life easier. You should move this out of the main function and declare this as a global helper function instead. This will make your main code block a lot shorter